### ISSUE TITLE
musl build failure

### ISSUE THREAD

**Issue by CameronNemo** [containers/fuse-overlayfs#145]
Need something like https://github.com/containers/fuse-overlayfs/pull/52 for plugin-manager.c.

https://github.com/containers/fuse-overlayfs/blob/34794cf17f9f6397926367ce58e8a76b8770c3d8/plugin-manager.c#L22

**Comment by giuseppe** [comment]
can you check if https://github.com/containers/fuse-overlayfs/pull/147 solves the problem for you?

**Comment by mgoltzsche** [comment]
@giuseppe Unfortunately #147 doesn't solve the issue. Now the build fails with:
```
direct.c:159: undefined reference to `TEMP_FAILURE_RETRY'
direct.c:131: undefined reference to `TEMP_FAILURE_RETRY'
direct.c:153: undefined reference to `TEMP_FAILURE_RETRY'
```
The `TEMP_FAILURE_RETRY` macro that was defined in `main.c` (#52) should also be made available in `direct.c` as you did with `error`.

**Comment by giuseppe** [comment]
@mgoltzsche would you like to open a PR?  Otherwise I am afraid I'll iterate again and we can hit another issue.  We probably need some CI tests to avoid these kind of regressions as well :-)

**Comment by mgoltzsche** [comment]
Apparently it has been fixed with #149 and #150 meanwhile. Thanks!
